File: 2021/day1/day1_test.go
package main

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestDay1(t *testing.T) {
	inputArr := []int{199, 200, 208, 210, 200, 207, 240, 269, 260, 263}
	result := Day1(inputArr)
	assert.Equal(t, 7, result, "they should be equal")
}

func TestDay1_2(t *testing.T) {
	inputArr := []int{199, 200, 208, 210, 200, 207, 240, 269, 260, 263}
	result := Day1_2(inputArr)
	assert.Equal(t, 5, result, "they should be equal")
}

File: 2021/day1/nodejs/day1.test.js
import assert from "assert";
import { day1Part1, day1Part2 } from "./day1.js";

describe("day 1 part 1", () => {
  it("case 1", () => {
    const inputArr = [199, 200, 208, 210, 200, 207, 240, 269, 260, 263];
    const result = day1Part1(inputArr);
    assert.equal(result, 7);
  });
});

describe("day 1 part 2", () => {
  it("case 1", () => {
    const inputArr = [199, 200, 208, 210, 200, 207, 240, 269, 260, 263];
    const result = day1Part2(inputArr);
    assert.equal(result, 5);
  });
});
